### ISSUE TITLE
Reproducing the Wan-1.3B-t2v bidirectional model

### ISSUE THREAD

**Issue by fwtan** [tianweiy/CausVid#7]
Hi the CausVid team, 

Thanks for the great work!

We're currently trying to reproduce the bidirectional wan-1.3b-t2v DMD model but are not quite there yet.

Specifically, we utilized the mixkit_latents_lmdb data provided at https://huggingface.co/tianweiy/CausVid/tree/main/mixkit_latents_lmdb and launched training using a script closely resembling the one at https://github.com/tianweiy/CausVid#training:

```
torchrun --nnodes 1 --nproc_per_node=8 causvid/train_distillation.py \
    --config_path  configs/wan_bidirectional_dmd_from_scratch.yaml 
```

We used 8 GPUs with a batch size of 4 per GPU, resulting in a global batch size of 32 (half of the original script). After training for 1400 iterations, the generated videos visualized in WandB appear entirely as mosaics.

May I if this behavior is expected given the smaller batch size or if we missed any important hyper-parameters?

Thank you for your help!

**Comment by tianweiy** [comment]
did you try any sampling scripts? The wandb might not support video logging yet

**Comment by wangshankun** [comment]
@fwtan 
  1 batchsize 1 gpu，使用了72G； 你是如何4 batchsize 1 gpu； 😁
<img width="458" alt="Image" src="https://github.com/user-attachments/assets/4849f855-83a3-4ab2-b98f-45c2d17ca0ac" />

**Comment by fwtan** [comment]
> e.g. https://github.com/tianweiy/CausVid/blob/master/minimal_inference/bidirectional_inference.py

Thanks, @tianweiy ! I will try it out soon and let you know tomorrow.

**Comment by fwtan** [comment]
> [@fwtan](https://github.com/fwtan) 1 batchsize 1 gpu，使用了72G； 你是如何4 batchsize 1 gpu； 😁 <img alt="Image" width="458" src="https://private-user-images.githubusercontent.com/3970271/426346873-4849f855-83a3-4ab2-b98f-45c2d17ca0ac.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDI4ODEzMTgsIm5iZiI6MTc0Mjg4MTAxOCwicGF0aCI6Ii8zOTcwMjcxLzQyNjM0Njg3My00ODQ5Zjg1NS04M2EzLTRhYjItYjk4Zi00NWMyZDE3Y2EwYWMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDMyNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAzMjVUMDUzNjU4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NzA5M2Y4OTQ4MzFkOTA5OWE3MGRmMzVkZWNjNmE5ZTkxOTgwNmZmMzNlMmFjNDlhOWQzNWFiZTYwMWVlYmVjOCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.YYrd1_R3NG-QfkTlGy30Tor_dOKIVVYOjeTSeJvAl84">

With the default hyper-parameters, we observed a gpu memory consumption of around 57G for batch size 1. Using a larger batch size didnt significantly increase the memory usage with gradient checkpointing

**Comment by fwtan** [comment]
> > e.g. https://github.com/tianweiy/CausVid/blob/master/minimal_inference/bidirectional_inference.py
> 
> Thanks, [@tianweiy](https://github.com/tianweiy) ! I will try it out soon and let you know tomorrow.

No luck. We used 4 Node x 8 GPU x Batch Size 1, the video generated by the bidirectional inference script was still not correct.

https://github.com/user-attachments/assets/5b47d13f-bc94-433e-9527-1ddffcfdb936

**Comment by tianweiy** [comment]
hi, not sure what modification you have done. I just used the master branch and trained one model on a single node and default parameter. It at least gets the content right . 

Wanda log https://wandb.ai/yin2/causvid/runs/8axhn7wn?nw=nwuseryin2 

https://github.com/user-attachments/assets/f3fd73ea-846f-406e-958a-cc938bfadcd8
https://github.com/user-attachments/assets/4f7f36e7-a54b-4534-95e7-dc3700888bcd

**Comment by fwtan** [comment]
> hi, not sure what modification you have done. I just used the master branch and trained one model on a single node and default parameter for a few hundred iterations. It at least gets the content right .
> 
> Wanda log https://wandb.ai/yin2/causvid/runs/8axhn7wn?nw=nwuseryin2
> 
> https://github.com/user-attachments/assets/f3fd73ea-846f-406e-958a-cc938bfadcd8 https://github.com/user-attachments/assets/4f7f36e7-a54b-4534-95e7-dc3700888bcd

Thanks, @tianweiy for the extra experiments!
It looks like the wandb log is private by default. Is there any chance to make it public so that we could check if our training curve matches yours? (may also be helpful to https://github.com/tianweiy/CausVid/issues/8#issue-2946471120)

**Comment by fwtan** [comment]
> https://wandb.ai/yin2/causvid/runs/8axhn7wn should work

Hi Tianwei,  Thanks again for running the extra experiments! 
I finally had some time to figure out the issue: I misunderstood one of the hyper-parameters, i.e. num_train_steps, which should be set to 1000. I learned that the example experiment converged after 1000 iterations and mistakenly treated num_train_steps as the total number of training iterations. I set it to a much higher value to keep the job running (to avoid it being killed when GPU utilization dropped). This was of course incorrect as it disrupted the timestep sampling.

**Comment by wangshankun** [comment]
> [@fwtan](https://github.com/fwtan) 请问下wan-1.3B的模型有训练成功吗，有没有碰到亮度颜色过饱和的问题呢？

随着时间帧越来越亮，好像是ar训练本身的问题吧？
